src/components/ui/BuyButton/BuyButton.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import { Button as UIButton } from '@faststore/ui'
+import type { ButtonProps } from '@faststore/ui'
+import { ShoppingCart } from 'phosphor-react'
+
+type Props = ButtonProps
+
+function BuyButton({ children, ...props }: Props) {
+  return (
+    <UIButton {...props}>
+      <ShoppingCart />
+      {children}
+    </UIButton>
+  )
+}
+
+export default BuyButton
```

Comment:
Shouldn't we create a generic IconButton here and pass the icon as prop?

```suggestion
import type { ReactNode } from 'react'
import React from 'react'
import { Button as UIButton } from '@faststore/ui'
import type { ButtonProps } from '@faststore/ui'

type Props = ButtonProps & {
  icon: ReactNode
}

function IconButton({ children, icon, ...props }: Props) {
  return (
    <UIButton {...props}>
      {icon}
      {children}
    </UIButton>
  )
}

export default IconButton
```

Reply:
As far as I remember, this BuyButton should be a special button. So I think this is the reason why we aren't using a generic `IconButton`.

Reply:
Yep, this is a button that should only be used on the PDP for the buy action.